app/models/classification.rb

```diff
@@ -10,14 +10,18 @@ class Classification < ApplicationRecord
   before_destroy :delete_tags_and_entries
 
   validates :description, :presence => true, :length => {:maximum => 255}
+  # for :description, this tries to mimick :scope => [:region_id, :parent_id]
   validates :description, :uniqueness => {:scope => [:parent_id]}, :if => proc { |c|
     cond = c.class.in_region(region_id).where(:parent_id => c.parent_id, :description => c.description)
     cond = cond.where.not(:id => c.id) unless c.new_record?
-    cond.exists?
+    c.description_changed? && cond.exists?
   }
 
   NAME_MAX_LENGTH = 50
-  validates :name, :presence => true, :length => {:maximum => NAME_MAX_LENGTH}
+  validates :name, :presence => true, :length => {:maximum => NAME_MAX_LENGTH}, :if => proc { |c|
+    c.new_record? || name_changed?
```

Comment:
should this use `c.name_changed?` ?  is c == self ?

Reply:
`c` == `self`.

`name` is from/stored in the tag.
So looking up the name cost a query.

If it is a new record, fetching and verifying the name is good.
If it is not a new record, then we can assume that the tag is still valid (well, unless we changed the name)

Always validating the `name` means always fetching the tag field.